lib/rspec/core/notifications.rb

```diff
@@ -359,7 +361,11 @@ def formatted_load_time
       # @return [String] The summary information fully formatted in the way that
       #   RSpec's built-in formatters emit.
       def fully_formatted(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
-        formatted = "\nFinished in #{formatted_duration} " \
+        formatted = ""
+        if syntax_highlighting_unavailable
+          formatted << "\nSyntax highlighting of failure snippets unavailable -- install the coderay gem to enable it.\n"
```

Comment:
> It's a hard balance between advertising and getting out of the way!
> 
> We'd have to add a config flag for it, which seems possibly overkill.

Yep, I thought about the config flag, too, but yeah, that feels like overkill.  We do already kinda have a config flag, anyway: `--formatter`: you can configure a custom formatter that avoids displaying that message.  That'd be way overkill as well, though.

One other thing to bear in mind: the syntax highlighter is only attempted to be used if there's a failure, which means `reporter.syntax_highlighting_unavailable` is only set if there's a failure, which means that when there are no failures you won't get this message.  That's intentional since I figure the user isn't missing out when everything passes.  Hopefully that will make it less annoying for your case.